### PR TITLE
Fix mean_values for Saabas method

### PR DESCRIPTION
**Pull request by nalzok**
Address https://github.com/dmlc/xgboost/issues/8251

### CODE REVIEW

**Comment by nalzok** [comment]
Looks like I have passed all tests except the following one

https://github.com/dmlc/xgboost/blob/8d247f0d64bee392857b330b20bfe6086798eb7b/tests/cpp/predictor/test_cpu_predictor.cc#L74-L86

I'm not sure what's going on here. Is that because I assumed `base_weight` has been populated before calling `PredictContribution`, but it's not in this case?

File: src/predictor/cpu_predictor.cc [hunk 1]
```diff
@@ -274,13 +274,23 @@ float FillNodeMeanValues(RegTree const *tree, bst_node_t nidx, std::vector<float
...
     return;
   }
   mean_values->resize(num_nodes);
-  FillNodeMeanValues(tree, 0, mean_values);
+
+  if (approximate) {
+    // prediction value as if the node is a leaf
+    auto &node_mean_values = *mean_values;
+    for (bst_node_t nidx = 0; nidx < num_nodes; ++nidx) {
+      node_mean_values[nidx] = tree->Stat(nidx).base_weight;
```
Comment (RAMitchell) [inline src/predictor/cpu_predictor.cc:288]: I think `tree->Stat(nidx).base_weight` is not adjusted for learning rate. So this prediction would be much too large.

Comment (nalzok) [reply src/predictor/cpu_predictor.cc:288]: That's true, but unfortunately, we cannot access the training parameters (i.e. `learning_rate`) here. Do you have a better idea than the following hack?

https://github.com/dmlc/xgboost/pull/2003/commits/87d96d2c13663122346216c8eba0829dfdcbf175#diff-0c9f721872945f814f6d5e9ade9cd184b7e1882ce6a138f183f5af07c8520c65R596-R600

**Comment by trivialfis** [comment]
@RAMitchell Do you think there's a better approach?

**Comment by RAMitchell** [comment]
I think the current implementation is correct as we don't claim to implement the Saabas method.
